torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -289,9 +291,27 @@ def __init__(
         cpu_offload: Optional[CPUOffload] = None,
         auto_wrap_policy: Optional[Callable] = None,
         backward_prefetch: Optional[BackwardPrefetch] = None,
+        ignored_modules: Optional[Iterable[torch.nn.Module]] = None,
     ):
         torch._C._log_api_usage_once("torch.distributed.fsdp")
         super().__init__()
+        # Save the ignored modules and their parameters, including the
+        # parameter names, which are needed to filter the model state dict
+        self._ignored_modules = self._check_ignored_modules(ignored_modules)
+        ignored_params = set(p for m in self._ignored_modules for p in m.parameters())
+        param_to_unflat_param_names = _get_param_to_unflat_param_names(module)
+        self._ignored_param_to_param_name = {}
+        for param in ignored_params:
+            unflat_param_names = param_to_unflat_param_names[param]
+            if len(unflat_param_names) != 1:
```

Comment:
Can we just simplify this to `if len != 1: raise ValueError(...)`?

Reply:
The `assert` is meant as an internal check. If it is violated, then there may be a case for which `_get_param_to_unflat_param_names()` is incorrect. It might be good to keep the `assert`.